rita/src/rita_client/dashboard/mod.rs

```diff
@@ -156,3 +162,75 @@ impl Handler<SetWifiConfig> for Dashboard {
         Ok(())
     }
 }
+
+#[derive(Serialize)]
+pub struct NodeInfo {
+    pub nickname: String,
+    pub route_metric_to_exit: u16,
+    pub total_payments: i64,
+    pub debt: i64,
+    pub link_cost: u16,
+    pub price_to_exit: u32,
+}
+
+pub struct GetNodeInfo;
+
+impl Message for GetNodeInfo {
+    type Result = Result<Vec<NodeInfo>, Error>;
+}
+
+impl Handler<GetNodeInfo> for Dashboard {
+    type Result = ResponseFuture<Vec<NodeInfo>, Error>;
+
+    fn handle(&mut self, _msg: GetNodeInfo, _ctx: &mut Self::Context) -> Self::Result {
+        Box::new(
+            DebtKeeper::from_registry()
+                .send(Dump {})
+                .and_then(|res| {
+                    let stream = TcpStream::connect::<SocketAddr>(
+                        format!("[::1]:{}", SETTING.get_network().babel_port)
+                            .parse()
+                            .unwrap(),
+                    ).expect("Failed to connect to Babel!");
+                    let mut babel = Babel::new(stream);
+                    babel.start_connection().expect("Bad copy of Babel!");
+
+                    let res = res.unwrap();
+
+                    let mut output = Vec::new();
+
+                    let exit_client = SETTING.get_exit_client();
+                    let current_exit = exit_client.get_current_exit();
+
+                    for (identity, debt_info) in res.iter() {
+                        if current_exit.is_some() {
+                            let exit_ip = current_exit.unwrap().id.mesh_ip;
+                            let route = babel
+                                .get_route_via_neigh(identity.mesh_ip, exit_ip)
+                                .expect("Failed to find route for Neighbor!");
+                            output.push(NodeInfo {
+                                nickname: serde_json::to_string(&identity.mesh_ip).unwrap(),
+                                route_metric_to_exit: route.metric,
+                                total_payments: debt_info.total_payment_recieved.clone().into(),
+                                debt: debt_info.debt.clone().into(),
+                                link_cost: route.refmetric,
+                                price_to_exit: route.price,
+                            })
+                        } else {
+                            output.push(NodeInfo {
+                                nickname: serde_json::to_string(&identity.mesh_ip).unwrap(),
+                                route_metric_to_exit: 0,
+                                total_payments: debt_info.total_payment_recieved.clone().into(),
+                                debt: debt_info.debt.clone().into(),
+                                link_cost: 0,
+                                price_to_exit: 0,
```

Comment:
these should probrobly be something other than 0, probably an Option (as price_to_exit _might_ be 0)

Reply:
How does that make a difference here?

Reply:
Tells the JS that this is an invalid value, so it can not show it on the UI

Reply:
so return 404? that's probably better I agree.

Reply:
@kingoflolz so I played around with this some more, the types just really don't work out with the box's. Any ideas? I notice you did the same thing in wifi settings as I'm doing here so I'm pretty sure you didn't find a better way back then.